assets/js/modules/thank-with-google/datastore/publications.test.js

```diff
@@ -348,5 +352,67 @@ describe( 'modules/thank-with-google publications', () => {
 				);
 			} );
 		} );
+
+		describe( 'getServiceCreatePublicationURL', () => {
+			it( 'returns an account chooser URL with the home URL appended to the publisher center URL', () => {
+				provideSiteInfo( registry );
+				provideUserInfo( registry );
+
+				const homeURL = registry.select( CORE_SITE ).getHomeURL();
+				const publisherCenterURL = new URL(
+					'https://publishercenter.google.com/'
+				);
+				publisherCenterURL.searchParams.set(
+					'sk_url',
+					encodeURIComponent( homeURL )
+				);
+				const expectedAccountChooserURL = registry
+					.select( CORE_USER )
+					.getAccountChooserURL( publisherCenterURL );
+
+				const createPublicationURL = registry
+					.select( MODULES_THANK_WITH_GOOGLE )
+					.getServiceCreatePublicationURL();
+
+				expect( createPublicationURL ).toEqual(
+					expectedAccountChooserURL
+				);
+			} );
+		} );
+
+		describe( 'getServicePublicationURL', () => {
+			it( 'should throw an error if no publicationID is given', () => {
+				expect( () =>
+					registry
+						.select( MODULES_THANK_WITH_GOOGLE )
+						.getServicePublicationURL()
+				).toThrow( 'A publicationID is required.' );
+			} );
+
+			it( 'returns a publisher center URL for an existing publication', () => {
+				registry
+					.dispatch( MODULES_THANK_WITH_GOOGLE )
+					.receiveGetPublications(
+						publicationWithOnboardingCompleteState
+					);
+
+				registry
+					.dispatch( MODULES_THANK_WITH_GOOGLE )
+					.setPublicationID( 'test-publication-a' );
+
+				const publication = registry
+					.select( MODULES_THANK_WITH_GOOGLE )
+					.getCurrentPublication();
```

Comment:
Why do we need it here? Seems like we don't use neither the current publication nor publications in the `getServicePublicationURL` selector, right?

Reply:
You're right. I think I was overthinking this one. I've removed them and simplified it.

Reply:
No worries, all good 👍

Reply:
Oups wrong comment 😅